plugins/module_utils/syslog_server.py

```diff
@@ -0,0 +1,189 @@
+# -*- coding: utf-8 -*-
+# Copyright: (c) 2023, XLAB Steampunk <steampunk@xlab.si>
+#
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+from __future__ import annotations
+
+__metaclass__ = type
+
+from .rest_client import RestClient
+
+# from .client import Client
+from ..module_utils.utils import PayloadMapper, get_query
+from ..module_utils.task_tag import TaskTag
+
+from ..module_utils.typed_classes import (
+    TypedTaskTag,
+    TypedSyslogServerToAnsible,
+    TypedSyslogServerFromAnsible,
+)
+from typing import Union, Any, Dict
+
+
+class SyslogServer(PayloadMapper):
+    def __init__(
+        self,
+        uuid: Union[str, None] = None,
+        alert_tag_uuid: Union[str, None] = None,
+        host: Union[str, None] = None,
+        port: Union[int, None] = None,
+        protocol: Union[str, None] = None,
+        resend_delay: Union[int, None] = None,
+        silent_period: Union[int, None] = None,
+        latest_task_tag: Union[TypedTaskTag, dict[Any, Any], None] = None,
+    ):
+        self.uuid = uuid
+        self.alert_tag_uuid = alert_tag_uuid
+        self.host = host
+        self.port = port
+        self.protocol = protocol
+        self.resend_delay = resend_delay
+        self.silent_period = silent_period
+        self.latest_task_tag = latest_task_tag if latest_task_tag is not None else {}
+
+    @classmethod
+    def from_ansible(cls, ansible_data: TypedSyslogServerFromAnsible):
+        return SyslogServer(
+            uuid=ansible_data["uuid"],
+            alert_tag_uuid=ansible_data["alert_tag_uuid"],
+            host=ansible_data["host"],
+            port=ansible_data["port"],
+            protocol=ansible_data["protocol"],
+        )
+
+    @classmethod
+    def from_hypercore(cls, hypercore_data: dict[Any, Any]):
+        if not hypercore_data:
+            return None
+        return cls(
+            uuid=hypercore_data["uuid"],
+            alert_tag_uuid=hypercore_data["alertTagUUID"],
+            host=hypercore_data["host"],
+            port=hypercore_data["port"],
+            protocol=hypercore_data["protocol"],
+            resend_delay=hypercore_data["resendDelay"],
+            silent_period=hypercore_data["silentPeriod"],
+            latest_task_tag=hypercore_data["latestTaskTag"],
+        )
+
+    def to_hypercore(self) -> dict[Any, Any]:
+        return dict(
+            alertTagUUID=self.alert_tag_uuid,
+            host=self.host,
+            port=self.port,
+            protocol=self.protocol,
+            resendDelay=self.resend_delay,
+            silentPeriod=self.silent_period,
```

Comment:
`alertTagUUID`, `resendDelay`, `silentPeriod` - likely we should treat them as readonly HC3 API attributes.

Checking what webUI sends to HC3. websocket message was:
```
42["command",{"action":"update","type":"AlertSyslogTarget","commandID":"b08b8c33-d815-406c-84e7-7bfe53644aa4","data":{"host":"10.5.11.222","port":"514","protocol":"0","uuid":"b08b8c33-d815-406c-84e7-7bfe53644aa4"}},["authPassword","sharedSecret"]]
```
Confirmed, on update/patch those values are not sent to HC3.